### PR TITLE
Issue 12396 to add developer documentation to readme.

### LINKED ISSUE #12396: Add ./mach run -Z help under command line arguments in README
There are a bunch of nice developer flags when running servo with ./mach run -Z but this isn't listed anywhere in README.me here - https://github.com/servo/servo#commandline-arguments

It'd be nice if it was documented.

### PR DESCRIPTION
**Pull request by changm**
Just some added developer documentation that was useful for me while starting to develop servo. 

This adds the documentation about running ./mach clean with a partially built servo but before linking openssl.
## It also adds the -Z flag under command line flags for debugging help.

- [X ] `./mach build -d` does not report any errors
- [ X] `./mach test-tidy` does not report any errors
- [ X] These changes fix #12396  (github issue number if applicable).

- [ ] There are tests for these changes OR
- [X] These changes do not require tests because it is a documentation update only.

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/12514)

### CODE REVIEW

**Comment by SimonSapin** [comment]
Looks good, thanks!

@bors-servo r+ rollup

File: README.md [hunk 1]
```diff
@@ -176,6 +178,7 @@ URL with servo).
   `INTERVAL` seconds
 - `-s SIZE` sets the tile size for painting; defaults to 512
 - `-z` disables all graphical output; useful for running JS / layout tests
+- `-z help` displays useful output to debug servo
```
Comment (jdm) [inline README.md]: Shouldn't this be a capital Z?

Comment (SimonSapin) [inline README.md]: Oh, that’s right.

**Comment by SimonSapin** [comment]
@bors-servo r-

Please change to capital Z per @jdm’s comment. Thanks!

**Comment by changm** [comment]
Changed, thanks for spotting that.

**Comment by changm** [comment]
Hmm the compiler segfaulted from the log here, even though this commit didn't touch any actual code. Sorry I'm unfamiliar with the process here, what happens next? I didn't see a way to re=run AppVeyor anywhere. Thanks!

**Comment by jdm** [comment]
That's being tracked in #12512.

**Pull request merged**